tools/ci_build/build.py

```diff
@@ -1977,6 +1977,11 @@ def run_nodejs_tests(nodejs_binding_dir):
     run_subprocess(args, cwd=nodejs_binding_dir)
 
 
+def install_python_wheel_build_deps():
+    dep_packages = ["setuptools>=41.4.0", "wheel==0.37.1"]
```

Comment:
- `setuptools>=41.4.0` is specified in requirements-dev.txt and in all of our /ci_build/github/linux/docker/scripts/* scripts.
- `wheel` is not stable, so pinning this to latest working version 0.37.1.

Reply:
Actually, this function is not used by our packaging pipelines, and I don't want to introduce it. We use requirements.txt to manage such things. And different OS/Python version may need different versions of the same package. 

Our component governance tool can recognize requirements.txt , but not this.

Reply:
Thanks for looking at this. I suspected that this was not the best way to handle this. I can close this draft to allow someone else to apply the correct fix.

Reply:
> Actually, this function is not used by our packaging pipelines, and I don't want to introduce it. We use requirements.txt to manage such things. And different OS/Python version may need different versions of the same package.
> 
> Our component governance tool can recognize requirements.txt , but not this.

requirements.txt has the correct version dependency updated.
@adrianlizarraga 's concern was for users who build a python package on their local machine. how do we fix it for them?

Reply:
Maybe we can add a switch in build.py. If it is on, call the "install_python_wheel_build_deps" function, if is off, don't do that. And our linux pipelines do not need it. Because the pipelines run in docker. And we install python packages when building the docker images. And build.py runs after images are built.  This is because usually we don't need to call pip install for every PR. As long as the build environment is not changed, we can cache the result. 

And, in order to reduce inconsistency, it's better to only call pip install once. Otherwise the second call may override something in the first call.  For example, if in the first call you set numpy==1.16.6. Then you call "pip install" again to install some other packages, for example, scipy. Then the second one may upgrade numpy to a newer version and it's hard to notice.

Reply:
I'm getting this error when trying to build the CUDA docker image. Any progress on a fix?